account-kit/smart-contracts/src/ma-v2/modules/single-signer-validation/signer.ts

```diff
@@ -49,7 +53,49 @@ export const singleSignerMessageSigner = (
   entityId: number,
   deferredActionData?: Hex
 ) => {
+  const signingMethods: SigningMethods = {
+    prepareSign: async (
+      request: SignatureRequest
+    ): Promise<SignatureRequest> => {
+      let ret: SignatureRequest = {
+        type: "eth_signTypedData_v4",
+        data: {
+          domain: {
+            chainId: Number(chain.id),
+            verifyingContract:
+              getDefaultSingleSignerValidationModuleAddress(chain),
+            salt: concatHex([`0x${"00".repeat(12)}`, accountAddress]),
+          },
+          types: {
+            ReplaySafeHash: [{ name: "hash", type: "bytes32" }],
+          },
+          message: {
+            hash: "",
+          },
+          primaryType: "ReplaySafeHash",
+        },
+      };
+
+      if (request.type === "personal_sign") {
+        ret.data.message.hash = hashMessage(request.data);
+      } else if (request.type === "eth_signTypedData_v4") {
+        ret.data.message.hash = hashTypedData(request.data);
+      } else {
+        throw new Error(`Unsupported signature request type`);
+      }
+
+      return ret;
```

Comment:
nit: this is a pretty opinionated style thing, but i think it would be more clear to just do it like this (so everything is immutable):

```suggestion
      const hash = request.type === "personal_sign"
        ? hashMessage(request.data)
        : request.type === "eth_signTypedData_v4"
        ? hashTypedData(request.data)
        : assertNever(request.typem "Unsupported signature request type")

      return {
        type: "eth_signTypedData_v4",
        data: {
          domain: {
            chainId: Number(chain.id),
            verifyingContract:
              getDefaultSingleSignerValidationModuleAddress(chain),
            salt: concatHex([`0x${"00".repeat(12)}`, accountAddress]),
          },
          types: {
            ReplaySafeHash: [{ name: "hash", type: "bytes32" }],
          },
          message: {
            hash,
          },
          primaryType: "ReplaySafeHash",
        },
      };
```

Reply:
i like this better too, will make the change